### DISCUSSION TITLE
Confusion Regarding Course Selections

### DISCUSSION THREAD

**Discussion by Tanush103** [snigi-gupta/AttendingUniversityAtBuffalo#10] [General]
Hi Snigdha,
Thanks for the webinar ...it was very helpful.I was having aone query and need your help.
Like If I want to take 2 courses from same focus area which contains introductory courses ..the focus areas from which we are allowed to take one per FA ...For Example Taking both Introduction to Machine Learning as well to Introduction to Computer Vision and Image Processing.....Can we do that  ??

**Comment by snigi-gupta** [comment]
Hi @Tanush103,
I would suggest you to go through the CS grad handbook and my readme file which is specific for CS. 
If you have already read it and still need clarification, then let me explain in simpler terms:
- You will have to actually take 3 courses from the same focus area. For eg: if I am planning to take ML/AI as my focus area, then I need 1 breadth course (lets say Intro to ML), and 2 depth courses (lets say 1 600 level course like Robot learning, 1 500 or 600 level course like CVIP)

In short, you can and you might have to take both ML and CVIP if you want to. 

Please read point number three in the below screenshot
![image](https://user-images.githubusercontent.com/56351901/112261590-932f1a80-8c42-11eb-9aa2-9f5291a80f79.png)

Link to the page: [here](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/tree/main/Computer%20Science%20Important%20Tips)

**Reply by Tanush103** [reply]
Okay...So I can take a more than 1 500 level course from the same breadth right ???If suppose I'm not going for AI Specialization just wanted to take 1 compulsory course (Intro to ML) and one additional course apart from depth one (CVIP)

**Reply by snigi-gupta** [reply]
@Tanush103 Did you read the entire link I shared? Also the graduate handbook?

**Reply by Tanush103** [reply]
Okay I think I got it now thanks

**Reply by snigi-gupta** [reply]
@Tanush103 ,

So after you have chosen everything according to the ms requirement checklist, you will have 6 credits remaining which means you can take literally anything worth of 6 credits. This means you can take an additional ML or AI related course or an additional seminar or independent study, a course from SWE or hardware breadth. Infact you can also take a course from another department if you get the permission from the graduate coordinator and graduate director. 
Like you can also take a course from Data Science, Music, or Management. 

The last 6 credits is something you have full flexibility to use for any type of course. This is also, that in case you were not able to full fill your ms requirement by the last sem (3rd sem) you still have the opportunity to fix that and graduate. You need to full fill your ms requirement to graduate.

**Reply by snigi-gupta** [reply]
@Tanush103,

So apart from the 24 credits, you do for the MS requirement, you have the freedom to choose any course from any department for 6 credits. For example, let's say your specialization is Software, and you completed all the requirements, you have the freedom to take two 3-credit courses from any department, or even take a seminar. You can take a course in data science, management, music, etc. Those remaining 6 credits are yours to choose from in any way. Although you would need the permission from graduate coordinator and graduate director to pick a course outside of CS department if you can justify how learning that course is beneficial to your degree. I wanted to take Project Management, but due to workload balance, I was unable to.

Also, sometimes, due to improper planning of courses or unforeseen situations (failing a course, not fulfilling the requirements or academic integrity case, or etc) at times some students are unable to complete the requirements and then they can use these 6 credits to choose a course specifically that fulfills a requirement. 
For eg: let's say you got a D in algos and because of that your breadth GPA is below 3.0, then you will have to take another algo subject to fix that GPA.

I hope this clears your doubt.

**Reply by Tanush103** [reply]
Yes got it ...Thanks a lot Snigdha for your help..😊😊
